### ISSUE TITLE
Support for Shelly Plus H&T

### ISSUE THREAD

**Issue by retrry** [esphome/feature-requests#2030]
**Describe the problem you have/What new integration you would like**
I would like to have support for Shelly Plus H&T device. It is based on ESP32 processor.

**Please describe your use case for this integration and alternatives you've tried:**
I would like to use these devices for temperature monitoring in my home with Home Assistant.

**Additional context**
https://www.shelly.cloud/en/products/shop/shelly-plus-h-and-t
https://kb.shelly.cloud/knowledge-base/shelly-plus-h-t
I can buy couple devices for a person, who will implement support for it in ESPHome.

**Comment by avilleret** [comment]
I guess the main issue here is the E-paper graphic display that needs to be retro engineered to understand how it is wired and how to control it with custom firmware.
The rest should be an ESP32 platform with hygrometry and temperature sensors which should be not so difficult to interface with ESPHome.

**Comment by Robbe-B** [comment]
I'm really willing to give up the display of that device in favour of having faster sensor updates from the custom firmware.

I was planning on using this device to control humidity inside my bathroom by triggering the bathroom fan. (e.g. during a rapid humidity increase when taking a shower)

But unfortunately this device (out of the box) is just too slow for this. The fastest updates you get (when powering via usb) is each 5 minutes.

I'm really hoping that this can change when we flash this device with esphome. 
Even is the display is nog working anymore, I would be happy.

**Comment by grischard** [comment]
Shelly [claims](https://www.reddit.com/r/ShellyUSA/comments/1aofmrd/anyone_having_trouble_with_shelly_ht_gen3_staying/kqf4mhz/) that this is necessary so that the radio doesn't heat up the thermometer.

**Comment by Robbe-B** [comment]
> Shelly [claims](https://www.reddit.com/r/ShellyUSA/comments/1aofmrd/anyone_having_trouble_with_shelly_ht_gen3_staying/kqf4mhz/) that this is necessary so that the radio doesn't heat up the thermometer

@grischard , I'm aware of this, would this also be the case for the newer Gen3 models ?

**Comment by grischard** [comment]
@Robbe-B yes. While esphome would be cool on these (the e-ink display looks nice), they probably only nerf the thermometer and updates because they have no other choice. Overall, it feels like a poorly designed device that's full of quirks.

**Comment by Robbe-B** [comment]
> @Robbe-B yes. While esphome would be cool on these (the e-ink display looks nice), they probably only nerf the thermometer and updates because they have no other choice. Overall, it feels like a poorly designed device that's full of quirks.

@grischard, 
I totally agree, 
I was kinda disappointed when I discovered all this after buying it.
I even bought the Gen3 because they promised faster updates. 
But really, nothing changed compared to the Gen2 models.

**Comment by mstipanov** [comment]
I have Shelly BLU H&T Display ZB and using this. Works fine:

sensor:
  - platform: template
    name: "Kitchen Temperature"
    id: kitchen_temperature
    unit_of_measurement: "°C"
    accuracy_decimals: 1
    device_class: "temperature"
    state_class: "measurement"
    update_interval: never

  - platform: template
    name: "Kitchen Humidity"
    id: kitchen_humidity
    unit_of_measurement: "%"
    accuracy_decimals: 1
    device_class: "humidity"
    state_class: "measurement"
    update_interval: never

  - platform: template
    name: "Kitchen Battery Level"
    id: kitchen_battery_level
    unit_of_measurement: "%"
    accuracy_decimals: 0
    device_class: "battery"
    state_class: "measurement"
    update_interval: never

button:
  - platform: restart
    name: "Restart"

esp32_ble_tracker:
  on_ble_advertise:
    - mac_address: "xx:xx:xx:xx:xx:xx"
      then:
        - lambda: |-
            ESP_LOGD("ble_adv", "Received advertisement from %s", x.address_str().c_str());
            for (auto &service_data : x.get_service_datas()) {
              if (service_data.uuid.contains(0xD2, 0xFC)) { // 0xFCD2 (BTHome)
                auto data = service_data.data;
                ESP_LOGD("bthome", "Found BTHome data: %s", format_hex_pretty(data).c_str());
                // BTHome V2 parsing
                // Byte 0: Control (e.g., 0x40)
                for (size_t i = 1; i < data.size(); ) {
                  uint8_t type = data[i++];
                  if (type == 0x00 && i < data.size()) { // Packet ID (1 byte)
                    i += 1;
                  } else if (type == 0x01 && i < data.size()) { // Battery (1 byte, 1%)
                    float batt = data[i];
                    id(kitchen_battery_level).publish_state(batt);
                    ESP_LOGD("bthome", "Battery: %.0f%%", batt);
                    i += 1;
                  } else if (type == 0x02 && i + 1 < data.size()) { // Temperature (2 bytes, 0.01°C)
                    float temp = (int16_t(uint16_t(data[i+1]) << 8 | data[i])) * 0.01f;
                    id(kitchen_temperature).publish_state(temp);
                    ESP_LOGD("bthome", "Temperature: %.2f", temp);
                    i += 2;
                  } else if (type == 0x03 && i + 1 < data.size()) { // Humidity (2 bytes, 0.01%)
                    float hum = (uint16_t(uint16_t(data[i+1]) << 8 | data[i])) * 0.01f;
                    id(kitchen_humidity).publish_state(hum);
                    ESP_LOGD("bthome", "Humidity: %.2f", hum);
                    i += 2;
                  } else if (type == 0x45 && i + 1 < data.size()) { // Temperature (2 bytes, 0.1°C)
                    float temp = (int16_t(uint16_t(data[i+1]) << 8 | data[i])) * 0.1f;
                    id(kitchen_temperature).publish_state(temp);
                    ESP_LOGD("bthome", "Temperature: %.1f", temp);
                    i += 2;
                  } else if (type == 0x2E && i < data.size()) { // Humidity (1 byte, 1%)
                    float hum = data[i];
                    id(kitchen_humidity).publish_state(hum);
                    ESP_LOGD("bthome", "Humidity: %.0f", hum);
                    i += 1;
                  } else if (type == 0x1E && i < data.size()) { // Button (1 byte)
                    i += 1;
                  } else {
                    ESP_LOGD("bthome", "Unknown or truncated BTHome type: 0x%02X at index %d", type, i-1);
                    break; 
                  }
                }
              }
            }
